src/pkg/artifact/dao/dao.go

```diff
@@ -189,6 +193,27 @@ func (d *dao) Update(ctx context.Context, artifact *Artifact, props ...string) e
 	}
 	return nil
 }
+
+func (d *dao) UpdatePullTime(ctx context.Context, id int64, pullTime time.Time) error {
+	ormer, err := orm.FromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	// can only be retained to the second if not format
+	formatPullTime := pullTime.Format("2006-01-02 15:04:05.999999")
+	// update db only if pull_time is null or pull_time < (in-coming)pullTime
+	sql := "UPDATE artifact SET pull_time = ? WHERE id = ? AND pull_time IS NULL OR pull_time < ?"
```

Comment:
`"UPDATE artifact SET pull_time = ? WHERE id = ? AND pull_time IS NULL OR pull_time < ?"` => 

`"UPDATE artifact SET pull_time = ? WHERE id = ? AND (pull_time IS NULL OR pull_time < ?)"`